test/requests/discover_specs_test.rb

```diff
@@ -0,0 +1,87 @@
+# typed: true
+# frozen_string_literal: true
+
+require "test_helper"
+
+module RubyLsp
+  class DiscoverSpecsTest < Minitest::Test
+    def test_discovers_top_level_specs
+      source = File.read("test/fixtures/minitest_spec_simple.rb")
+
+      with_minitest_spec_configured(source) do |items|
+        assert_equal(["BogusSpec"], items.map { |i| i[:label] })
+      end
+    end
+
+    def test_discovers_nested_specs
+      source = File.read("test/fixtures/minitest_spec_nested.rb")
+
+      with_minitest_spec_configured(source) do |items|
+        top_level_specs = items[0][:children]
+        assert_equal(
+          ["First Spec"],
+          top_level_specs.map { |i| i[:label] },
+        )
+
+        nested_specs = top_level_specs[0][:children]
+        assert_equal(
+          ["test one", "test two"],
+          nested_specs.map { |i| i[:label] },
+        )
+      end
+    end
+
+    def test_discovers_dynamic_spec_names
+      source = File.read("test/fixtures/minitest_spec_dynamic_name.rb")
+
+      with_minitest_spec_configured(source) do |items|
+        nested_specs = items[0][:children][0][:children]
+        assert_equal(
+          ["dynamic_name"],
+          nested_specs.map { |i| i[:label] },
+        )
+      end
+    end
+
+    def test_handles_empty_specs
+      source = File.read("test/fixtures/minitest_spec_simple.rb")
+
+      with_minitest_spec_configured(source) do |items|
+        nested_specs = items[0][:children][0][:children]
+        assert_empty(
```

Comment:
```suggestion
        assert_empty(nested_specs)
```
(no point mapping over it if empty)